### ISSUE TITLE
plugin manager should not mention or consider incompatible versions of plugins to install

### ISSUE THREAD

**Issue by talevy** [elastic/logstash#5279]
Logstash should ignore versions of plugins that are not compatible with the current version of Logstash.

In the following case, I am asked whether I would like to install logstash-output-kafka 4.0, which is only compatible with LS 5.x. Then it ignores that recommendation and finds the largest version of the plugin that _is_ compatible, and installs that properly.

- Version: 2.3.x
- Operating System: All
- Steps to Reproduce:

1. install logstash 2.3.1
2. run `bin/logstash-plugin update logstash-output-kafka`

```
$ bin/logstash-plugin update logstash-output-kafka
You are updating logstash-output-kafka to a new version 4.0.0, which may not be compatible with 2.0.3. are you sure you want to proceed (Y/N)?
y
Updating logstash-output-kafka

WARNING: SSLSocket#session= is not supported
Updated logstash-codec-json 2.1.3 to 2.1.4
Updated logstash-output-kafka 2.0.3 to 2.0.5
```

**Comment by ph** [comment]
We need to make https://github.com/ph/logstash/blob/c02635a81a06aadfeff5c368b3654783108379e9/lib/pluginmanager/util.rb#L52-L59 a bit more intelligent and only fetch compatible gems with the current LS.

**Comment by purbon** [comment]
I agree with @ph, however doing this check more intelligent will require us to interact more close with how rubygems and bundler does version resolution, might be some bundler internals might help here?
